server/controller/src/main/java/ai/starwhale/mlops/domain/dataset/dataloader/DataReadManager.java

```diff
@@ -110,17 +118,19 @@ public Session generateSession(DataReadRequest request) {
     @Transactional
     public DataReadLog assignmentData(String consumerId, Session session) {
         var sid = session.getId();
+        var queue = sessionCache.computeIfAbsent(String.valueOf(sid), id -> new ConcurrentLinkedQueue<>());
 
-        // get first
-        var dataRange = dataReadLogDao.selectTop1UnAssignedData(sid);
-
-        if (Objects.nonNull(dataRange)) {
-            dataRange.setConsumerId(consumerId);
-            dataRange.setAssignedNum(dataRange.getAssignedNum() + 1);
-            dataReadLogDao.updateToAssigned(dataRange);
-            log.info("Assignment data id: {} to consumer:{}", dataRange.getId(), dataRange.getConsumerId());
+        if (queue.isEmpty()) {
+            queue.addAll(dataReadLogDao.selectTopsUnAssignedData(sid, cacheSize));
+        }
+        DataReadLog readLog = queue.poll();
```

Comment:
If an exception occurs later, how to deal with the items that have been taken out?

Reply:
this is unnecessary to deal it, it will be selected at the next round cache.